### PR TITLE
Clean up sfcsub.F: remove unused code, comments, and developer tags

### PR DESCRIPTION
**Pull request by ClaraDraper-NOAA**
As a first step to cleaning up the sfcsub routine, this PR removes obvious unused code and excessive commenting. It reduces the length of the code substantially, making it a little easier to parse/work with.

These edits were made by by co-pilot, with my reviewing them and adding some comments back in. 

These changes should not affect the functioning of the code in any way.

## Description of Changes:

Specific changes reported by co-pilot.
- Removed ~700 lines of excessive comments
- Removed 790 lines of isolated comment markers and excessive blank lines
- Removed 3 unused epsilon variables (epsoro, epsplr, epsacn)
- Removed 2 unused namelist variables (qcmsk, znlst)
- Removed 3 unused subroutines (count, dayoyr, maxmin)
- Removed 31 lines with developer suffix comments (cbosu, cggg, landice, cjfe)
- Removed all inline comments from routine argument lines
- Removed 7 commented-out old argument lines
- Removed 9 lines of commented-out old code (print statements and calculations)

Total reduction: ~1600 lines from original 8892 to 7275 lines

## Tests Conducted:

on gaea: 
Ran 4 cycles of the C96C48_hybatmDA global_workflow test case, confirmed that results are zero-diff. This will test the routine as called by both the model and UFS_UTILS. 

All ufs-weather-model tests passed on URSA.

on ursa: 
all UFS_UTILS/global_cycle tests passed. 

## Dependencies:
N/A

## Documentation:
N/A

## Issue (optional):

First step to addressing issue [#344](https://github.com/ufs-community/ccpp-physics/issues/344)

### CODE REVIEW

**Comment by ClaraDraper-NOAA** [comment]
draft while I do testing.

**Comment by ClaraDraper-NOAA** [comment]
@XuLi-NOAA Could you please review this PR?

I think the real issue is to make sure the gcycle is called at the initial time of the IAU, where the model forecast is involved. Then, to determine how often the gcycle will be called in the forecast mode. I think nscyc, a namelist parameter for model should be set as intended, probably larger than 6 hours (the IAU time window), which means gcycle is called only once (at the initial time). Importantly, we need to figure out why gcycle is called twice in IAU, which is supposedly unexpected. 
In summary, the content of this PR needs to be modified to address two issues: 1. Why gcycle is called twice?  2. Make sure gcycle is called at the initial time, not at other times. In other words, gcycle should be called only one time and at the initial time, unless it is designed so.

**Comment by ClaraDraper-NOAA** [comment]
Tests have passed, this is ready to review.

**Comment by ClaraDraper-NOAA** [comment]
> @XuLi-NOAA Could you please review this PR?
> 
> I think the real issue is to make sure the gcycle is called at the initial time of the IAU, where the model forecast is involved. Then, to determine how often the gcycle will be called in the forecast mode. I think nscyc, a namelist parameter for model should be set as intended, probably larger than 6 hours (the IAU time window), which means gcycle is called only once (at the initial time). Importantly, we need to figure out why gcycle is called twice in IAU, which is supposedly unexpected. In summary, the content of this PR needs to be modified to address two issues: 1. Why gcycle is called twice? 2. Make sure gcycle is called at the initial time, not at other times. In other words, gcycle should be called only one time and at the initial time, unless it is designed so.

@XuLi-NOAA I don't see the relevance of the timing of the global_cycle call to this PR. Does this comment relate to this [PR](https://github.com/NOAA-EMC/global-workflow/pull/4327) instead? Note that that PR has been approved and merged. No further work is required on it (but to respond to your comment: with IAU on, global_cycle is called on the sfc_data files once at the start of the DA window ( to be used to start the model) and again in the middle of the DA window. My understanding is that the output of the latter is needed by products. It is called twice, this is correct / intended.

**Comment by XuLi-NOAA** [comment]
> > @XuLi-NOAA Could you please review this PR?
> > I think the real issue is to make sure the gcycle is called at the initial time of the IAU, where the model forecast is involved. Then, to determine how often the gcycle will be called in the forecast mode. I think nscyc, a namelist parameter for model should be set as intended, probably larger than 6 hours (the IAU time window), which means gcycle is called only once (at the initial time). Importantly, we need to figure out why gcycle is called twice in IAU, which is supposedly unexpected. In summary, the content of this PR needs to be modified to address two issues: 1. Why gcycle is called twice? 2. Make sure gcycle is called at the initial time, not at other times. In other words, gcycle should be called only one time and at the initial time, unless it is designed so.
> 
> @XuLi-NOAA I don't see the relevance of the timing of the global_cycle call to this PR. Does this comment relate to this [PR](https://github.com/NOAA-EMC/global-workflow/pull/4327) instead? Note that that PR has been approved and merged. No further work is required on it (but to respond to your comment: with IAU on, global_cycle is called on the sfc_data files once at the start of the DA window ( to be used to start the model) and again in the middle of the DA window. My understanding is that the output of the latter is needed by products. It is called twice, this is correct / intended.

In principle, gcycle is called during a forecast mode, at the initial time 100% and how often for later on depends on the setting of one model namelist paramter. I understand IAU works with a model integration, even if itself is an analysis issue, and the available analysis increment is applied to the model integration when needed (I guess, in 6-hour IAU time window, this happens 6 times or houly). Therefore, the timing of gcycle is relavent, I understand.

File: physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F [hunk 1]
```diff
@@ -241,30 +227,24 @@ subroutine sfccycle(lugb,len,lsoil,sig1t,deltsfc                    &
 !     veg .. vegetation cover
 !     sot .. soil type
 !     soc .. soil color
-!cwu [+2l] add sih & sic
 !     sih .. sea ice thickness
 !     sic .. sea ice concentration
-!clu [+6l] add swd,slc,vmn,vmx,slp,abs
```
Comment (BrianCurtis-NOAA) [inline physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:247]: What was this old code comment signifying with [+61]

Comment (ClaraDraper-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:247]: I don't know. My guess is it added 61 lines.

File: physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F [hunk 2]
```diff
@@ -3642,72 +2827,32 @@ subroutine la2ga(regin,imxin,jmxin,rinlon,rinlat,rlon,rlat,inttyp,&
...
-!
-      return
-      end
-
 !>\ingroup mod_sfcsub
       subroutine filanl(tsfanl,tsfan2,wetanl,snoanl,zoranl,albanl,      &
      &                  aisanl,                                         &
      &                  tg3anl,cvanl ,cvbanl,cvtanl,                    &
      &                  cnpanl,smcanl,stcanl,slianl,scvanl,veganl,      &
-     &                  vetanl,sotanl,socanl,alfanl,                    & !socanl: soil color
```
Comment (BrianCurtis-NOAA) [inline physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:3696]: might want to keep this comment

Comment (BrianCurtis-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:3696]: Or maybe something like `! 'soc' refers to soil color`

File: physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F [hunk 3]
```diff
@@ -4909,17 +3945,9 @@ subroutine merge(len,lsoil,iy,im,id,ih,fh,deltsfc,                &
       if (fh-deltsfc < -0.001 .and. irttsf == 1) then
         rtsfs = 1.0
         rtsfl = 1.0
-!       do k=1,lsoil
-!         rsmcl(k) = 1.0
-!         rsmcs(k) = 1.0
-!         rstcl(k) = 1.0
-!         rstcs(k) = 1.0
-!       enddo
-      endif
```
Comment (BrianCurtis-NOAA) [inline physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:4918]: Are we chopping off an endif accidentally here?

Comment (BrianCurtis-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:4918]: I see how my git diff shows this, not removing endif, nvm

Comment (ClaraDraper-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:4918]: I tripped over this one too!

File: physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F (hunk 2 above)
Comment (ClaraDraper-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F:3696]: The variable naming is detailed at the top of the file, including: 

!     soc .. soil color

[here](https://github.com/ClaraDraper-NOAA/ccpp-physics/blob/b30b93b470dd453a83fb1b87b372fa327c2c9b24/physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F#L229C17-L229C24)

I don't think we need it defined again here.

**Comment by ClaraDraper-NOAA** [comment]
@BrianCurtis-NOAA Thanks for looking at this. I think I've addressed everything.

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F [hunk 4]
```diff
@@ -8746,12 +7126,13 @@ subroutine fixrda(lugb,fngrib,kpds5,slmask,                       &
      &               ' nearest matching dates (going back).'
           endif
         endif
-!
 !  no matching ih found. search nearest hour
-!
         if(ihr.gt.0.and.ihr.le.12) then
           ihr=0
           go to 50
+        elseif(ihr.eq.12) then
```
Comment (rhaesung) [inline physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F]: @ClaraDraper-NOAA Looks like this is redundant. Copilot probably added it by mistake?

Comment (ClaraDraper-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/sfcsub.F]: Thanks @rhaesung . That was actually me messing up a merge. I've fixed it.

**Review by rhaesung — APPROVED** [review]

**Comment by grantfirl** [comment]
@ClaraDraper-NOAA Do you plan to open supermodule PRs and run UFS regression tests on this?

**Review by grantfirl — APPROVED** [review]

**Comment by ClaraDraper-NOAA** [comment]
> @ClaraDraper-NOAA Do you plan to open supermodule PRs and run UFS regression tests on this?

@grantfirl Thanks for looking at this. 

Looping in @BrianCurtis-NOAA, as the sfcsub routine is also called by UFS_UTILS/global_cycle

Grant - this PR is zero diff for the output of sfscub (essentially it removes excessive comments). I have at least one more coming that will also be zero diff (removing code that cannot be exercised / is incomplete / very out-dated), and another that will be zero diff for the inputs that we use from ufs_model and UFS_UTILS (removing code that we are not currently exercising; simplifying code). I have two questions: 

1. How would your team like me to handle the supermodule PRs? Eventually there will need to be updates to the two calling routines (in UFS_UTILS and ccpp-physics), and associated ufs_model supermodules, and also to the global_cycle and ufs_utils scripts that prepare the namsfc namelist used by sfcsub.

My suggestion is that I construct all of the above PRs such that they do not affect the calling programs in any way, so that we do not need to do the supermodule updates. I have started collecting the namelist and argument changes (all deletions), and can then submit a fourth PR after the above three are merged that consists only of removing the namelist and argument variables from throughout the various modules / scripts. 

2. How would you like me to handle testing?  My suggestion is that I write a separate wrapper to test that my changes to sfcsub do not change any output for the range of possible functionality that we are using in either ufs_model or ufs_utils (as in it will explicitly test for changes in output from the different combinations of input variables). I will also do the standard UFS_UTILS/global_cycle tests (these check for zero diff change), and check two of the global_workflow tests are zero diff after cycling through one day (one test mimics GFSv17 and one mimics the uncoupled atmos model).

**Comment by grantfirl** [comment]
> > @ClaraDraper-NOAA Do you plan to open supermodule PRs and run UFS regression tests on this?
> 
> @grantfirl Thanks for looking at this.
> 
> Looping in @BrianCurtis-NOAA, as the sfcsub routine is also called by UFS_UTILS/global_cycle
> 
> Grant - this PR is zero diff for the output of sfscub (essentially it removes excessive comments). I have at least one more coming that will also be zero diff (removing code that cannot be exercised / is incomplete / very out-dated), and another that will be zero diff for the inputs that we use from ufs_model and UFS_UTILS (removing code that we are not currently exercising; simplifying code). I have two questions:
> 
> 1. How would your team like me to handle the supermodule PRs? Eventually there will need to be updates to the two calling routines (in UFS_UTILS and ccpp-physics), and associated ufs_model supermodules, and also to the global_cycle and ufs_utils scripts that prepare the namsfc namelist used by sfcsub.
> 
> My suggestion is that I construct all of the above PRs such that they do not affect the calling programs in any way, so that we do not need to do the supermodule updates. I have started collecting the namelist and argument changes (all deletions), and can then submit a fourth PR after the above three are merged that consists only of removing the namelist and argument variables from throughout the various modules / scripts.
> 
> 2. How would you like me to handle testing?  My suggestion is that I write a separate wrapper to test that my changes to sfcsub do not change any output for the range of possible functionality that we are using in either ufs_model or ufs_utils (as in it will explicitly test for changes in output from the different combinations of input variables). I will also do the standard UFS_UTILS/global_cycle tests (these check for zero diff change), and check two of the global_workflow tests are zero diff after cycling through one day (one test mimics GFSv17 and one mimics the uncoupled atmos model).

What is the reason for separating these 3 pieces of work into separate PRs? To make it easier to review due to the large number of changes?

If so, since this one has already been reviewed, perhaps you could add on to it and reviewers could review the next separately when it is ready? It is easy enough to review individual commits that come in if we want to avoid sifting through the already-completed changes.

I suppose that I would hold off on creating supermodule PRs in UFSATM, ufs-weather-model, and UFS-utils until all of the work with sfcsub.F is completed. Supermodule PRs are needed regardless of whether any code is changed within them or not. At the very least, they will just have submodule commit hash changes (and changes to .gitmodules for the automatic testing). 

Your testing plan sounds great, but we'll also have to run UFS regression tests on multiple platforms in order to merge it once you're satisfied with your own individual testing.

**Comment by ClaraDraper-NOAA** [comment]
Yes, I'm separating them as each will be rather large, and do slightly different things. I like your idea of adding everything into this PR. Then I'll do the full regression tests once everything is finalized.

**Review by Qingfu-Liu — APPROVED** [review]

**Comment by ClaraDraper-NOAA** [comment]
All ufs-weather-model tests have passed on URSA. Super module PRs submitted: 

* UFS WEATHER MODEL [PR](https://github.com/ufs-community/ufs-weather-model/pull/3076)

* UFSATM:  [PR](https://github.com/NOAA-EMC/ufsatm/pull/1062)

This is ready to be merged.

**Comment by rhaesung** [comment]
@ClaraDraper-NOAA Thanks for submitting these super module PRs and for confirming that all ufs-weather-model tests passed on URSA. 
For both PRs, please update the .gitmodules files to point to your working branches. For the UFS PR, also update the test_changes.list and URSA log files so the UFS code managers can refer to the results. Once these updates are made, the PRs will be ready to enter the commit queue.

**Comment by gspetro-NOAA** [comment]
Testing has completed successfully on WM PR 3072. This PR can be merged.

**Comment by ClaraDraper-NOAA** [comment]
> @ClaraDraper-NOAA Thanks for submitting these super module PRs and for confirming that all ufs-weather-model tests passed on URSA. For both PRs, please update the .gitmodules files to point to your working branches. For the UFS PR, also update the test_changes.list and URSA log files so the UFS code managers can refer to the results. Once these updates are made, the PRs will be ready to enter the commit queue.

@rhaesung As you probably realized, I was on leave yesterday. It looks like you've already sorted all of this out - thank you! Please let me know if you need anything else from me (note - I'm on leave again from tomorrow for 10 days).
